compiler/rustc_trait_selection/src/traits/error_reporting/on_unimplemented.rs

```diff
@@ -149,7 +149,19 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             .unwrap_or_else(|| (trait_ref.def_id(), trait_ref.skip_binder().substs));
         let trait_ref = trait_ref.skip_binder();
 
-        let body_hir = self.tcx.hir().local_def_id_to_hir_id(obligation.cause.body_id);
+        let body_def_id =
```

Comment:
It would be nice to comment why it makes sense to use `fn_def_id` here, I don't think it's immediately obvious. Also, style-wise, this could be a `match self.tcx.opt_rpitit_info` -- it doesn't need to be an if + match.

That being said, I also don't think it's correct to use `fn_def_id` here. We probably should just avoid adding an `ItemContext` flag, unless this is for diagnostics? But in that case, I would like some info on what tests breaks if you don't do this.

Reply:
Fixed.